spec/jobs/in_person/enrollments_ready_for_status_check/enrollment_pipeline_spec.rb

```diff
@@ -0,0 +1,334 @@
+require 'rails_helper'
+
+RSpec.describe InPerson::EnrollmentsReadyForStatusCheck::EnrollmentPipeline do
+  let(:error_reporter) { instance_double(InPerson::EnrollmentsReadyForStatusCheck::ErrorReporter) }
+  let(:email_body_pattern) { /\A\s*(?<enrollment_code>\d{16})\s*\Z/ }
+  subject(:enrollment_pipeline) { described_class.new(error_reporter:, email_body_pattern:) }
+
+  before(:each) do
+    allow(IdentityConfig.store).to receive(:in_person_enrollments_ready_job_email_body_pattern).
+      and_return('\A\s*(?<enrollment_code>\d{16})\s*\Z')
+
+    allow(error_reporter).to receive(:report_error)
+  end
+
+  describe '#process_message' do
+    let(:sqs_message) { instance_double(Aws::SQS::Types::Message) }
+    let(:sqs_message_id) { Random.uuid }
+    let(:sns_message_id) { Random.uuid }
+    let(:enrollment_code) { 16.times.map { rand(0..9) }.join }
+    let(:user) { create(:user) }
+    let(:user_id) { user.id }
+    let(:mail_date) { 16.hours.ago.to_datetime }
+    let(:ses_payload) do
+      {
+        content: Mail.new do |m|
+          m.text_part = enrollment_code
+        end.to_s,
+        mail: {
+          messageId: Random.uuid.delete('-'),
+          timestamp: DateTime.now.to_s,
+          source: 'testsource@example.com',
+          commonHeaders: {
+            date: Mail::DateField.new(mail_date).to_s,
+            messageId: Mail::Utilities.generate_message_id,
+          },
+        },
+      }
+    end
+    let(:logged_ses_values) do
+      {
+        ses_aws_message_id: ses_payload[:mail][:messageId],
+        ses_mail_source: ses_payload[:mail][:source],
+        ses_mail_timestamp: ses_payload[:mail][:timestamp],
+        ses_rfc_message_id: ses_payload[:mail][:commonHeaders][:messageId],
+        ses_rfc_origination_date: mail_date.to_s,
+      }
+    end
+    let(:sns_payload) do
+      {
+        MessageId: sns_message_id,
+        Message: ses_payload.to_json,
+      }
+    end
+    let(:expected_error) { nil }
+    let(:expected_error_extra) { nil }
+
+    before(:each) do
+      allow(sqs_message).to receive(:message_id).
+        and_return(sqs_message_id)
+    end
```

Comment:
Since `sqs_message` is doubling for a plain struct, could we just use the real value type instead?
```suggestion
    let(:sqs_message) { Aws::SQS::Types::Message.new(message_id: sqs_message_id) }
```

Reply:
This builds into the test some assumptions about how the struct is created, so I would prefer to avoid this change.

Reply:
Is the worry that the class constructor changes and breaks, or something else?

Reply:
@mitchellhenke Yes, roughly speaking. i.e. constructor or factory doing some kind of interpretation on the data passed in rather than setting it to exactly match the original parameters.

Reply:
There is an additional concern about coupling and misrepresenting parts of the payload that we don't actually use.